node/config/load.go

```diff
@@ -17,15 +18,40 @@ import (
 
 // FromFile loads config from a specified file overriding defaults specified in
 // the def parameter. If file does not exist or is empty defaults are assumed.
-func FromFile(path string, def interface{}) (interface{}, error) {
+func FromFile(path string, def interface{}, lc loadabilityCheck) (interface{}, error) {
+	// check for loadability
 	file, err := os.Open(path)
 	switch {
 	case os.IsNotExist(err):
+		if err := lc.configNotFound(); err != nil {
+			return nil, err
+		}
 		return def, nil
 	case err != nil:
 		return nil, err
 	}
+	cfgBs, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to read config for loadability checks %w", err)
+	}
+	s := string(cfgBs)
+	if err := lc.configOk(s); err != nil {
+		return nil, xerrors.Errorf("config failed loadability check: %w", err)
+	}
+	if err := file.Close(); err != nil {
+		return nil, err
+	}
 
+	file, err = os.Open(path)
```

Comment:
Why are we doing this twice?

Reply:
Mostly just me not knowing how computers work, I'll figure out how to reuse to make this nicer.